### DISCUSSION TITLE
Screen doesn't automatically redraw after calling ResetTerminal

### DISCUSSION THREAD

**Discussion by bavo96** [wezterm/wezterm#6398] [Q&A]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

After calling ResetTerminal, the screen is empty: 
![image](https://github.com/user-attachments/assets/c64a03a8-2cbd-4ee9-9a54-d4952f15dea3)
And after resizing or moving the cursor around, text appears again:
![image](https://github.com/user-attachments/assets/5e904aa0-78e6-4564-a63a-0701057b7123)
Is this an issue or did I miss any configuration ? 

My wezterm's configuration:

```lua
local wezterm = require 'wezterm'
local io = require 'io'

local os_name = io.popen("uname"):read('*l')

local function set_brightness()
    if os_name == 'Linux' then
        return 0.5 -- for Linux
    else
        return 0.6 -- for Darwin
    end
end

return {
    front_end = "WebGpu",
    prefer_egl = true,
    font = wezterm.font_with_fallback {
        'JetBrains Mono',
        'DejaVu Sans Mono',
        'DejaVu Sans',
    },
    color_scheme = 'GruvboxDarkHard',
    window_background_image = wezterm.home_dir .. '/.config/wezterm/batman.jpg',
    window_background_image_hsb = {
        -- Darken the background image by reducing it to 1/3rd
        brightness = set_brightness(),
        -- You can adjust the hue by scaling its value.
        -- a multiplier of 1.0 leaves the value unchanged.
        hue = 1.0,
        -- You can adjust the saturation also.
        saturation = 1.0,
    },
    window_background_opacity = 1,
    text_background_opacity = 1,
    font_size = 13,
    window_padding = {
        left = 0,
        right = 0,
        top = 0,
        bottom = 0,
    },
    keys = {
        -- Make Option-Left equivalent to Alt-b which many line editors interpret as backward-word
        { key = "LeftArrow",  mods = "OPT", action = wezterm.action { SendString = "\x1bb" } },
        -- Make Option-Right equivalent to Alt-f; forward-word
        { key = "RightArrow", mods = "OPT", action = wezterm.action { SendString = "\x1bf" } },
        -- Make Command-r equivalent to reset current wezterm terminal
        { key = "r",          mods = "CMD", action = wezterm.action.ResetTerminal },
        -- {key='C',mods='SHIFT|CTRL', action=wezterm.action.CopyTo 'ClipboardAndPrimarySelection', },
    },
    enable_scroll_bar = true,
    min_scroll_bar_height = '2cell',
    colors = {
        scrollbar_thumb = 'gray',
    },
    check_for_updates = false,
    check_for_updates_interval_seconds = 3600 * 24 * 7,
    harfbuzz_features = { 'calt=0', 'clig=0', 'liga=0' }
}

```

**Comment by bavo96** [comment]
Do you have any idea about this behavior @wez ?

**Reply by wez** [reply]
resetting the terminal resets the entire terminal emulation state. It's a one-sided operation that happens in the terminal. It doesn't communicate to the kernel about the reset because there isn't a defined way to do that. As such, there's no way for an application  to know that the terminal was reset and that it needs to redraw the screen.

You shouldn't normally need to reset the terminal, so it is unusual for it to be assigned to a keyboard short cut the way you have it in your config.

**Reply by bavo96** [reply]
Thanks for your advice, but I have a situation where I need to reset my terminal. After ssh to a server and a long while of not using the terminal (I close my laptop and open it again), the terminal is freezing and it takes me a while (like 5 mins) for wezterm to activate again. In that case, I think about using the `ResetTerminal` action. Do you think there is another better way to handle this situation?

**Reply by wez** [reply]
Do you know about: https://lonesysadmin.net/2011/11/08/ssh-escape-sequences-aka-kill-dead-ssh-sessions/ ?

**Reply by bavo96** [reply]
I mean at that state, the screen is freezing and I can't type any thing (even escape sequences for SSH (like Ctrl + C/Z))
